docs/developer/architecture.rst

```diff
@@ -246,6 +246,68 @@ documentation page on :doc:`contributing </contributing>`.
     - Uses the Ray_ execution framework.
     - The compute kernel/in-memory format is a pyarrow Table.
 
+Modin Implementation Details
+----------------------------
```

Comment:
Let's rename this section to "Module/Class View" as agreed off-github @amyskov

Reply:
Section was renamed.